src/Microsoft.AspNet.OData.Shared/Query/SkipTokenQueryOption.cs

```diff
@@ -0,0 +1,352 @@
+             * OR (true AND Prop1=Value1 AND Prop2=Value2 AND Prop3>Value3)
+             * and so on...
+             * Adding the first true to simplify implementation.
+             */
+            Expression lastEquality = Expression.Constant(true);
+            foreach (KeyValuePair<string, object> item in _propertyValuePairs)
+            {
+                string key = item.Key;
+                MemberExpression property = Expression.Property(param, key);
+                object value = item.Value;
+
+                Expression compare = null;
+                ODataEnumValue enumValue = value as ODataEnumValue;
+                if (enumValue != null)
+                {
+                    value = enumValue.Value;
+                }
+                Expression constant = parameterizeConstant ? LinqParameterContainer.Parameterize(value.GetType(), value) : Expression.Constant(value);
+                if (directionMap.ContainsKey(key))
+                {
+                    compare = directionMap[key] == OrderByDirection.Descending ? binder.CreateBinaryExpression(BinaryOperatorKind.LessThan, property, constant, true) : binder.CreateBinaryExpression(BinaryOperatorKind.GreaterThan, property, constant, true);
+                }
+                else
+                {
+                    compare = binder.CreateBinaryExpression(BinaryOperatorKind.GreaterThan, property, constant, true);
+                }
+
+                Expression condition = Expression.AndAlso(lastEquality, compare);
+                where = where == null ? condition : Expression.OrElse(where, condition);
+
+                lastEquality = Expression.AndAlso(lastEquality, binder.CreateBinaryExpression(BinaryOperatorKind.Equal, property, constant, true));
+                count++;
+            }
+
+            Expression whereLambda = Expression.Lambda(where, param);
+            return ExpressionHelpers.Where(query, whereLambda, query.ElementType);
+        }
+
+        private static IDictionary<string, OrderByDirection> PopulateDirections(OrderByQueryOption orderBy)
+        {
+            IDictionary<string, OrderByDirection> directions = new Dictionary<string, OrderByDirection>();
+            if (orderBy == null)
+            {
+                return directions;
+            }
+
+            foreach (OrderByPropertyNode node in orderBy.OrderByNodes)
+            {
+                if (node != null)
+                {
+                    directions[node.Property.Name] = node.Direction;
+                }
+            }
+            return directions;
+        }
+
+        /// <summary>
+        /// Bad Documentation
+        /// </summary>
+        /// <param name="result"></param>
+        /// <param name="model"></param>
+        /// <param name="orderByQueryOption"></param>
+        /// <returns></returns>
+        public static string GetSkipTokenValue(IQueryable result, IEdmModel model, OrderByQueryOption orderByQueryOption)
```

Comment:
>GetSkipTokenValue [](start = 29, length = 17)

We should provide a DI for customer to create their own Skiptokenvalue generating logic.

Reply:
I agree with that in the longer term vision but we can discuss if that has to be part of this PR or we will develop it later once people start using SkipToken

Reply:
Hello,
For what it is worth we have such a use case where we need to generate a different skiptoken and apply it to the query in a different way than currently implemented.

Reply:
@Galad Thanks for the feedback. This PR is outdated now, I would be pleased if you can take a look at #1723 . You would be able to inject your own implementation that derives from SkipTokenHandler class. Please let me know if you find any shortcomings with that.

Reply:
@KanishManuja-MS Thanks a lot, I will take a look